src/serialization/sb3.js

```diff
@@ -791,30 +819,58 @@ const deserializeFields = function (fields) {
     return obj;
 };
 
+const decodeBlock = function (block) {
```

Comment:
Notes from discussion: In the long run it seems likely that we'll need to do a two-pass approach to loading. In the first pass we'll initiate loading each extension that the project uses, then we'll wait for them all to finish loading, then we'll do a second pass to actually load the blocks. We don't need that just yet since internal extensions can load synchronously, but we'll need it once we start loading extensions asynchronously.